python/ray/air/config.py

```diff
@@ -149,31 +148,46 @@ class DatasetConfig:
     # True by default for the "train" dataset only.
     required: Optional[bool] = None
 
-    # Whether the dataset can be streamed into memory using pipelined reads.
-    # When enabled, get_dataset_shard() returns DatasetPipeline instead of Dataset.
-    # Note that streaming isn't enabled unless you set stream_window_size too.
-    # False by default for all datasets.
-    streamable: Optional[bool] = None
-
     # Whether to transform the dataset with the fitted preprocessor. This must be
     # enabled at least for the dataset that is fit.
     # True by default for all datasets.
     transform: Optional[bool] = None
 
-    # List of fields that the user cannot override in ``dataset_config``.
-    _noncustomizable_fields: Tuple[str] = ()
+    # Whether the dataset should be streamed into memory using pipelined reads.
+    # When enabled, get_dataset_shard() returns DatasetPipeline instead of Dataset.
+    # The amount of memory to use is controlled by `stream_window_size`.
+    # False by default for all datasets.
+    use_stream_api: Optional[bool] = None
 
-    # TODO: add stream_window_size, global/local shuffle options
+    # Configure the streaming window size in bytes. A good value is something like
+    # 20% of object store memory. If set to -1, then an infinite window size will be
+    # used (similar to bulk ingest). This only has an effect if use_stream_api is set.
+    # Set to 1.0 GiB by default.
+    stream_window_size: Optional[float] = None
+
+    # Whether to enable global shuffle (per pipeline window in streaming mode). Note
+    # that this is an expensive all-to-all operation, and most likely you want to use
+    # local shuffle instead.
+    # False by default for all datasets.
+    global_shuffle: Optional[bool] = None
+
+    # Validator function for the final config, which should raise ValueError if the
+    # config cannot be validated.
+    _validator: Optional[Callable[["DatasetConfig"], None]] = None
```

Comment:
I feel like you may need to support a list of validators, like:
```
_validators=[DisallowUsage("use_stream_api"), HuggingFaceCfgValidation()]
```
another option I think is to tell users to override the validate():
```
class HuggingFaceTrainer:
  def validate(...):
    super().validate(...)
    # More custom validation I'd like to add.
```

Reply:
Ok yeah, just doing it in validate() is much better, changed!